### PR TITLE
Add FullPageNotFoundView

### PR DESCRIPTION
**Pull request by arosiclair**
Refactors `FullPageOfflineBlockingView` to use a shared `BlockingView` component and adds `FullPageNotFoundView` for use when the user navigates to a resource that doesn't exist or they have lost access to. 

The not found view replaces the content when the `shouldShow` prop is `true`.

### Fixed Issues

$ https://github.com/Expensify/App/issues/9522

### Tests

1. Wrap the `ScrollView` in [InitialSettingsPage.js](https://github.com/Expensify/App/blob/3757142611c84d9ffc59f86c29d1cd6f3db07f01/src/pages/settings/InitialSettingsPage.js) with the new `FullPageNotFoundView`
2. Set the `shouldShow` prop to `false` or omit it.
3. Verify you can open and view the normal settings page
4. Set the `shouldShow` prop to `true`
5. Verify the settings page is replaced with the not found blocking view

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

Same as Tests

- [x] Verify that no errors appear in the JS console

#### Web
Screen Shot 2022-07-05 at 3 08 50 PM

#### Mobile Web
Screenshot_20220705-154833

#### Desktop
Screen Shot 2022-07-05 at 3 11 23 PM

#### iOS
Screen Shot 2022-07-05 at 3 18 50 PM

#### Android

Screen Shot 2022-07-08 at 9 21 35 AM

### CODE REVIEW

**Comment by arosiclair** [comment]
I'm still waiting on text from marketing (and translation) for this page but the code is ready for review in the meantime

cc @Luke9389 @roryabraham since you guys we're on the PR for the blocking offline page

File: src/components/BlockingViews/FullPageNotFoundView.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const propTypes = {
+    /** Props to fetch translation features */
+    ...withLocalizePropTypes,
+
+    /** Child elements */
+    children: PropTypes.node.isRequired,
+
+    /** If true, child components are replaced with a blocking "not found" view */
+    show: PropTypes.bool,
```
Comment (roryabraham) [inline src/components/BlockingViews/FullPageNotFoundView.js:17]: This should either be required or have a default

**Review by roryabraham — CHANGES_REQUESTED** [review]
Otherwise LGTM 👍

**Comment by arosiclair** [comment]
Updated with default props and finalized text for english and spanish!

File: src/components/BlockingViews/BlockingView.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import variables from '../../styles/variables';
+import Icon from '../Icon';
+import Text from '../Text';
+
+const propTypes = {
+    /** Expensicon for the page */
+    icon: PropTypes.func.isRequired,
+
+    /** Color for the icon (should be from theme) */
+    iconColor: PropTypes.string.isRequired,
```
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.js:14]: NAB – seems like we could default this to `themeColors.offline` ?

File: src/components/BlockingViews/FullPageNotFoundView.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (roryabraham) [inline src/components/BlockingViews/FullPageNotFoundView.js:17]: Oh, we should call this `shouldShow` since it's a boolean

**Review by roryabraham — CHANGES_REQUESTED** [review]
Just some very minor comments, overall LGTM 👍

File: src/components/BlockingViews/BlockingView.js (hunk 2 above)
Comment (arosiclair) [reply src/components/BlockingViews/BlockingView.js:14]: Yeah I think that's fair. I'll add that in

**Review by roryabraham — APPROVED** [review]
LGTM 👍

**Comment by arosiclair** [comment]
Not sure how this is triggered, but this PR was tested on all platforms minus Android.

**Comment by Justicea83** [comment]
why not android?

**Comment by arosiclair** [comment]
Android build is currently broken on M1. @Justicea83 You don't have an intel mac you could test this with do you?

**Comment by Justicea83** [comment]
Nope, I use M1

**Comment by arosiclair** [comment]
Alright I was able to get the android build on my other laptop. Tested and added a screenshot.

@Justicea83 can you do internal QA to verify and check off this PR on the deploy checklist?

File: src/languages/en.js [hunk 4]
```diff
@@ -507,6 +507,8 @@ export default {
         chatYouLookingForCannotBeFound: 'The chat you are looking for cannot be found.',
         getMeOutOfHere: 'Get me out of here',
         iouReportNotFound: 'The payment details you are looking for cannot be found.',
+        notHere: "Hmm... it's not here",
```
Comment (Luke9389) [inline src/languages/en.js:510]: Would any of you be willing to link me to a convo where we decided on this copy?
Seems pretty uninformative to me.

**Comment by roryabraham** [comment]
I should have noticed that the QA steps don't work here. The testing steps require making code changes that we can't make on staging

**Comment by roryabraham** [comment]
Since this PR just creates some new components but does not use them anywhere, I'm tempted to say it's `[No QA]`?

File: src/languages/en.js (hunk 4 above)
Comment (Luke9389) [reply src/languages/en.js:510]: Ah, here: https://github.com/Expensify/App/issues/9522#issuecomment-1171702598
👍

**Comment by arosiclair** [comment]
Oh I didn't know that InternalQA meant explicitly testing on staging. But yeah with the ad-hoc testing me and Justice have done so far I think this is good to go with no further QA

**Pull request merged**